### PR TITLE
Logger final

### PR DESCRIPTION
**Pull request by jah701**

### CODE REVIEW

File: src/main/java/com/dev/cinema/Main.java [hunk 1]
```diff
@@ -54,28 +58,25 @@ public static void main(String[] args) throws AuthenticationException,
...
+        movieSessionService.findAvailableSessions(1L, LocalDate.now()).forEach(logger::info);
 
+        logger.info("Starting 'Authentication Service' test . . .");
         AuthenticationService authenticationService =
                 (AuthenticationService) injector.getInstance(AuthenticationService.class);
         User bob = authenticationService.register("bob@gmail.com", "qwerty123456");
-        System.out.println(authenticationService.login("bob@gmail.com", "qwerty123456"));
-
-        System.out.println("CART");
+        logger.info(authenticationService.login("bob@gmail.com", "qwerty123456"));
```
Comment (DmGrt) [inline src/main/java/com/dev/cinema/Main.java]: Better to wrap this in try-catch and add log.warn in the catch block

File: src/main/java/com/dev/cinema/dao/AbstractDao.java [hunk 2]
```diff
@@ -26,6 +28,7 @@ public T add(T entity, Class<T> clazz) {
             if (session != null) {
                 session.close();
             }
+            logger.info("Entity was added successfully: " + entity);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/AbstractDao.java]: better move this to `try` block

File: src/main/java/com/dev/cinema/Main.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/Main.java]: I agree

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/Main.java [hunk 3]
```diff
@@ -64,19 +66,33 @@ public static void main(String[] args) throws AuthenticationException,
         AuthenticationService authenticationService =
                 (AuthenticationService) injector.getInstance(AuthenticationService.class);
         User bob = authenticationService.register("bob@gmail.com", "qwerty123456");
-        logger.info(authenticationService.login("bob@gmail.com", "qwerty123456"));
+        String email = null;
+        try {
+            logger.info(authenticationService.login(email = "bob@gmail.com", "qwerty123456"));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java:68]: 1. fix travis
2. why catching Exeption? this is too abstract. you should be able to remove `throws AuthenticationException` from method signature

File: src/main/java/com/dev/cinema/Main.java [hunk 4]
```diff
@@ -64,19 +66,33 @@ public static void main(String[] args) throws AuthenticationException,
...
+        }
 
         logger.info("Starting 'Shopping Cart Service' test . . .");
         ShoppingCartService shoppingCartService =
                 (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
         shoppingCartService.addSession(movieSession, bob);
-        logger.info(shoppingCartService.getByUser(bob));
+        try {
+            logger.info(shoppingCartService.getByUser(bob));
+        } catch (Exception e) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: which exception do you expect from this method?

File: src/main/java/com/dev/cinema/Main.java [hunk 5]
```diff
@@ -64,19 +66,33 @@ public static void main(String[] args) throws AuthenticationException,
...
 
         logger.info("Starting 'Order Service' test . . .");
         OrderService orderService = (OrderService) injector.getInstance(OrderService.class);
         orderService.completeOrder(shoppingCartService.getByUser(bob).getTickets(), bob);
 
         List<Order> orderList = orderService.getOrderHistory(bob);
-        orderList.forEach(logger::info);
+        try {
+            orderList.forEach(logger::info);
+        } catch (Exception e) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/Main.java (hunk 3 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/Main.java:68]: Не уделил должного внимания бедному Main'у 😄

File: src/main/java/com/dev/cinema/Main.java [hunk 6]
```diff
@@ -79,19 +77,18 @@ public static void main(String[] args) throws AuthenticationException,
         shoppingCartService.addSession(movieSession, bob);
         try {
             logger.info(shoppingCartService.getByUser(bob));
-        } catch (Exception e) {
+        } catch (DataProcessingException e) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: let's not catch RuntimeException

File: src/main/java/com/dev/cinema/Main.java [hunk 7]
```diff
@@ -79,19 +77,18 @@ public static void main(String[] args) throws AuthenticationException,
...
-
         logger.info("Starting 'Order Service' test . . .");
         OrderService orderService = (OrderService) injector.getInstance(OrderService.class);
         orderService.completeOrder(shoppingCartService.getByUser(bob).getTickets(), bob);
 
         List<Order> orderList = orderService.getOrderHistory(bob);
         try {
             orderList.forEach(logger::info);
-        } catch (Exception e) {
+        } catch (DataProcessingException e) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**